### ISSUE TITLE
Adding convenience constructors for LogDetEstimator class

### ISSUE THREAD

**Issue by lambday** [shogun-toolbox/shogun#1932]
Currently [LogDetEstimator](http://shogun-toolbox.org/doc/en/latest/classshogun_1_1CLogDetEstimator.html) class uses a three argument constructor which takes
- A trace sampler
- A linear operator function
- A computation engine

This task is for hiding all these details from the front end by adding a couple of constructors for convenience, which would just take a SGMatrix or SGSparseMatrix and initialize everything with defaults, and then one should be able to sample the log-det estimates.

In short, we want something like this -

log_det_estimator=LogDetEstimator(A.tocsc()) near [this code](https://github.com/lambday/shogun/blob/develop/examples/undocumented/python_modular/mathematics_logdet.py#L71)

Also, unit tests for these is desired to make sure that things work properly. Should be easy :)

Please do not hesitate to ask @karlnapf or me regarding any doubts.
